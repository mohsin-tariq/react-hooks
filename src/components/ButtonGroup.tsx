import styled from 'styled-components'

const ButtonGroup = styled.div`
  display: flex;
  button:not(:last-child) {
    margin-right: 1rem;
  }
`

export default ButtonGroup
