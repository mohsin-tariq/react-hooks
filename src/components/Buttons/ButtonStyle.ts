import { css } from 'styled-components'
import { breakpoints } from '../tools'

export const PointStyles = css`
  &:hover {
    cursor: pointer;
  }

  &:active {
    cursor: pointer;
  }

  &:disabled {
    cursor: not-allowed;
  }
`

const ButtonStyle = css`
  display: flex;
  align-items: center;
  min-height: 4rem;
  padding: 0 2rem;
  font-size: 1.4rem;
  line-height: 1.8rem;
  font-weight: normal;

  svg {
    width: 2rem;
    height: 2rem;
    margin-right: 0.8rem;
  }

  ${breakpoints.lg} {
    width: auto;
  }
  ${breakpoints.sm} {
    width: 100%;
    justify-content: center;
  }
  ${PointStyles}
`

export default ButtonStyle
