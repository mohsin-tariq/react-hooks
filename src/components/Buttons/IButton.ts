export default interface ButtonInterface {
  readonly small?: boolean
  readonly children: any
}
