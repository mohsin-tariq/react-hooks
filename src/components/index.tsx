export * from './Buttons'
export * from './Timezone'
export * from './Card'
export { default as Container } from './Container'
export { default as Column } from './Column'
export { default as ButtonGroup } from './ButtonGroup'
export { default as Paragraph } from './Paragraph'
export { default as Title } from './Title'
export { default as Label } from './Label'
