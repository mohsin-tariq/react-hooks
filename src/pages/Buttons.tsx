import {
  PrimaryButton,
  SecondaryButton,
  TertiaryButton,
  Container,
  Column,
} from '../components'

const Buttons = (): JSX.Element => (
  <div>
    <Container>
      <Column>
        <h1>Normal</h1>
      </Column>
      <Column>
        <h1>Small</h1>
      </Column>
      <Column>
        <h1>Disabled</h1>
      </Column>
    </Container>
    <Container>
      <Column>
        <PrimaryButton type="button">Primary CTA</PrimaryButton>
      </Column>
      <Column>
        <PrimaryButton type="button" small>
          Primary CTA
        </PrimaryButton>
      </Column>
      <Column>
        <PrimaryButton type="button" disabled>
          Primary CTA
        </PrimaryButton>
      </Column>
    </Container>
    <Container>
      <Column>
        <SecondaryButton type="button">Secondary CTA</SecondaryButton>
      </Column>
      <Column>
        <SecondaryButton type="button" small>
          Secondary CTA
        </SecondaryButton>
      </Column>
      <Column>
        <SecondaryButton type="button" disabled>
          Secondary CTA
        </SecondaryButton>
      </Column>
    </Container>
    <Container>
      <Column>
        <TertiaryButton type="button">Tertiary CTA</TertiaryButton>
      </Column>
      <Column>
        <TertiaryButton type="button" small>
          Tertiary CTA
        </TertiaryButton>
      </Column>
      <Column>
        <TertiaryButton type="button" disabled>
          Tertiary CTA
        </TertiaryButton>
      </Column>
    </Container>
    <Container style={{ backgroundColor: '#555555', padding: '2rem 0' }}>
      <Column>
        <TertiaryButton type="button" light>
          Tertiary(L) CTA
        </TertiaryButton>
      </Column>
      <Column>
        <TertiaryButton type="button" light small>
          Tertiary(L) CTA
        </TertiaryButton>
      </Column>
      <Column>
        <TertiaryButton type="button" disabled light>
          Tertiary(L) CTA
        </TertiaryButton>
      </Column>
    </Container>
  </div>
)

export default Buttons
