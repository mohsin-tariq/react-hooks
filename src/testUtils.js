
/** eslint-disable **/
import { render as rtlRender } from '@testing-library/react'
import { Provider } from 'react-redux'
import configureMockStore from 'redux-mock-store'
import '@testing-library/jest-dom/extend-expect'

const initialState = {
  zone: {
    data: [
      {
        countryCode: 'AD',
        countryName: 'Andorra',
        zoneName: 'Europe/Andorra',
        gmtOffset: 3600,
        timestamp: 1614084530,
      },
      {
        countryCode: 'AL',
        countryName: 'Albania',
        zoneName: 'Europe/Albania',
        gmtOffset: 3600,
        timestamp: 1614084530,
      },
    ],
  },
}

const mockStore = configureMockStore()

function render(
  ui,
  { store = mockStore(initialState), ...renderOptions } = {}
) {
  // eslint-disable-next-line react/prop-types
  const Wrapper = ({ children }) => {
    return <Provider store={store}>{children}</Provider>
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions })
}

// re-export everything
export * from '@testing-library/react'
// override render method
export { render }
