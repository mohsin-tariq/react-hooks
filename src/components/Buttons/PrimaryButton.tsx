import styled, { css } from 'styled-components'
import ButtonStyle from './ButtonStyle'
import SmallButtonStyle from './SmallButtonStyle'
import { TertiaryDisabledStyle } from './TertiaryButton'
import IButton from './IButton'

const PrimaryButton = styled.button<IButton>`
  ${({ small }) =>
    css`
      ${small ? SmallButtonStyle : ButtonStyle}
    `}
  color: #FFFFFF;
  background-color: #db0011;
  border: none;
  &:hover {
    background-color: #a4000d;
  }
  &:active {
    background-color: #83000a;
  }
  &:disabled {
    ${TertiaryDisabledStyle}
  }
`

export default PrimaryButton
