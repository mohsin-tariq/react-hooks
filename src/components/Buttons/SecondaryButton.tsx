import styled, { css } from 'styled-components'
import ButtonStyle from './ButtonStyle'
import SmallButtonStyle from './SmallButtonStyle'
import { TertiaryDisabledStyle } from './TertiaryButton'
import IButton from './IButton'

const SecondaryButton = styled.button<IButton>`
  ${({ small }) =>
    css`
      ${small ? SmallButtonStyle : ButtonStyle}
    `}
  color: #FFFFFF;
  background-color: #4d6474;
  border: none;

  &:hover {
    background-color: #3e505c;
  }
  &:active {
    background-color: #31404a;
  }
  &:disabled {
    ${TertiaryDisabledStyle}
  }
`

export default SecondaryButton
