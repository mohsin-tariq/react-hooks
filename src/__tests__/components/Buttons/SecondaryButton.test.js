import { render, screen } from '../../../testUtils'
import { SecondaryButton } from '../../../components/Buttons'


describe('Secondary', () => {
  it('render Snapshot ', () => {
    const { asFragment } = render(<SecondaryButton type="button">Secondary</SecondaryButton>)
    expect(asFragment()).toMatchSnapshot()
  })

  it('Normal Button: verify background color, height and padding set to it ', () => {
    render(<SecondaryButton type="button">Secondary</SecondaryButton>)
    expect(screen.getByText('Secondary')).toHaveStyle('background-color: #4d6474')
    expect(screen.getByText('Secondary')).toHaveStyle('min-height: 4rem')
    expect(screen.getByText('Secondary')).toHaveStyle('padding: 0 2rem')
  })

  it('Small Button: verify background color, height and padding set to it ', () => {
    render(<SecondaryButton type="button" small>Secondary</SecondaryButton>)
    expect(screen.getByText('Secondary')).toHaveStyle('background-color: #4d6474')
    expect(screen.getByText('Secondary')).toHaveStyle('min-height: 3rem')
    expect(screen.getByText('Secondary')).toHaveStyle('padding: 0 1rem')
  })

  it('Disabled: verify background color', () => {
    render(<SecondaryButton type="button" disabled>Secondary</SecondaryButton>)
    expect(screen.getByText('Secondary').closest('button')).toBeDisabled()
    expect(screen.getByText('Secondary')).toHaveStyle('color: #252525')
    expect(screen.getByText('Secondary')).toHaveStyle('background-color: #fff')
    expect(screen.getByText('Secondary')).toHaveStyle('cursor: not-allowed')
  })
})