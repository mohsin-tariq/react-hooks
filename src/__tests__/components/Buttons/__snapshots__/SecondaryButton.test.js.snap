// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Secondary render Snapshot  1`] = `
<DocumentFragment>
  <button
    class="sc-dlfnbm jcNQPz"
    type="button"
  >
    Secondary
  </button>
</DocumentFragment>
`;
